### ISSUE TITLE
"WARNING: doo's init function was not set" is misleading

### ISSUE THREAD

**Issue by thedavidmeister** [bensu/doo#122]
```
(defn ^:export run! [a]
  (try
    (*main-cli-fn* a)
    (catch :default e
      (println "WARNING: doo's init function was not set")
      (println e)
      (exit! false))))
```

This just means that literally any error will print this warning, it does not mean that the init function wasn't set.

**Comment by bensu** [comment]
Hi,

Thanks for the report. Can you show me an example of an error and its cause that was unrelated to the message? It would be helpful to decide where/if to put that message.

**Comment by thedavidmeister** [comment]
@bensu I literally just had an error in my own code. The cause was a mistake i made.

**Comment by bensu** [comment]
Hi,

Great! Can you give me a small project that shows that error?

**Comment by lsenta** [comment]
https://xkcd.com/979/

**Comment by bensu** [comment]
Hi @Isenta

Would you mind providing a repo that shows the error?

**Comment by lsenta** [comment]
@bensu  https://github.com/bensu/doo/issues/134

**Comment by thedavidmeister** [comment]
this looks like an example https://github.com/bensu/doo/pull/141

**Comment by MatthewDarling** [comment]
In terms of improving the messaging here, is there more than `(println e)` that is specific to the actual problem that happened? I tried `(println (.-stack e))` but I don't know if that works in all environments.

I've opened #151 to avoid printing the "init function was not set" when it doesn't apply. But if there's other low-hanging fruit to improve the default exception handler, now's a good time to address that :)
